### PR TITLE
Added endpoints to read organizations

### PR DESCRIPTION
**Pull request by clintonb**
ECOM-6501

### CODE REVIEW

**Comment by clintonb** [comment]
@edx/ecommerce please review

**Comment by clintonb** [comment]
@edx/ecommerce please review

File: course_discovery/apps/api/v1/tests/test_views/test_organizations.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        organizations[1].tags.add(tag)
+        self.assert_list_tag_filter(Organization.objects.all(), [tag, tag2])
+
+    def test_retrieve(self):
+        """ Verify the endpoint returns details for a single organization. """
+        organization = OrganizationFactory()
+        url = reverse('api:v1:organization-detail', kwargs={'uuid': organization.uuid})
+
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, 200)
```
Comment (davec-edx) [inline course_discovery/apps/api/v1/tests/test_views/test_organizations.py:108]: Do these cover both positive and negative? I see checking for 200s for example. Is it possible to get anything else?

Comment (clintonb) [reply course_discovery/apps/api/v1/tests/test_views/test_organizations.py:108]: If you passed an invalid UUID, you'd get a 404. Other than that the only other codes would be 401/403 for auth (which is tested) and 500 for unexpected errors.

I can add a test for the 404.

**Review by schenedx — APPROVED** [review]
👍

File: course_discovery/apps/api/v1/views/organizations.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
+from rest_framework import viewsets
+from rest_framework.filters import DjangoFilterBackend
+from rest_framework.permissions import IsAuthenticated
+
+from course_discovery.apps.api import filters, serializers
+
+
+# pylint: disable=no-member
+class OrganizationViewSet(viewsets.ReadOnlyModelViewSet):
```
Comment (rlucioni) [inline course_discovery/apps/api/v1/views/organizations.py:9]: @clintonb what does this mean for the "organizations API" hosted by the LMS? Is it deprecated in favor of this viewset?

Comment (clintonb) [reply course_discovery/apps/api/v1/views/organizations.py:9]: Long term, it probably should be. In the near term, we will have two APIs; although, the API hosted by the Catalog Service will include info from Drupal.

**Pull request merged**